File: CRUD/Java usando NetBeans/Aplicacion Web usando MVC Spring V2/SistemaWebSpring/src/main/java/sistema/models/Proveedor.java
// Written by Ismael Heredia in the year 2017
package sistema.models;

import javax.validation.constraints.NotNull;
import org.hibernate.validator.constraints.NotEmpty;

public class Proveedor {

    int id;

    @NotEmpty
    String nombre;

    @NotEmpty
    String direccion;

    @NotNull
    int telefono;

    String fecha_registro;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public int getTelefono() {
        return telefono;
    }

    public void setTelefono(int telefono) {
        this.telefono = telefono;
    }

    public String getFecha_registro() {
        return fecha_registro;
    }

    public void setFecha_registro(String fecha_registro) {
        this.fecha_registro = fecha_registro;
    }

    public Proveedor() {

    }

    public Proveedor(int id, String nombre, String direccion, int telefono, String fecha_registro) {
        this.id = id;
        this.nombre = nombre;
        this.direccion = direccion;
        this.telefono = telefono;
        this.fecha_registro = fecha_registro;
    }

    @Override
    public String toString() {
        return "Proveedor{" + "id=" + id + ", nombre=" + nombre + ", direccion=" + direccion + ", telefono=" + telefono + ", fecha_registro=" + fecha_registro + '}';
    }

}

File: CRUD/PHP/Ajax V3/entities/Proveedor.php
<?php

class Proveedor
{
    
    private $id;
    private $nombre;
    private $direccion;
    private $telefono;
    private $fecha_registro;
    
    public function __construct()
    {
        $this->id   = "";
        $this->nombre = "";
        $this->direccion      = "";
        $this->telefono       = "";
        $this->fecha_registro = "";
    }
    
    public static function CreateProveedor($id, $nombre, $direccion, $telefono, $fecha_registro)
    {
        $instance                 = new self();
        $instance->id   = $id;
        $instance->nombre = $nombre;
        $instance->direccion      = $direccion;
        $instance->telefono       = $telefono;
        $instance->fecha_registro = $fecha_registro;
        return $instance;
    }
    
    public function getId()
    {
        return $this->id;
    }
    
    public function setId($id)
    {
        $this->id = $id;
    }
    
    public function getNombre()
    {
        return $this->nombre;
    }
    
    public function setNombre($nombre)
    {
        $this->nombre = $nombre;
    }
    
    public function getDireccion()
    {
        return $this->direccion;
    }
    
    public function setDireccion($direccion)
    {
        $this->direccion = $direccion;
    }
    
    public function getTelefono()
    {
        return $this->telefono;
    }
    
    public function setTelefono($telefono)
    {
        $this->telefono = $telefono;
    }
    
    public function getFecha_registro()
    {
        return $this->fecha_registro;
    }
    
    public function setFecha_registro($fecha_registro)
    {
        $this->fecha_registro = $fecha_registro;
    }
    
    public function __destruct()
    {
        $this->id   = "";
        $this->nombre = "";
        $this->direccion      = "";
        $this->telefono       = "";
        $this->fecha_registro = "";
    }
    
    public function toString()
    {
        return "Proveedor{" . "id=" . $this->id . ", nombre=" . $this->nombre . ", direccion=" . $this->direccion . ", telefono=" . $this->telefono . ", fecha_registro=" . $this->fecha_registro . '}';
    }
    
}

?>